src/Engine/Renderer/Renderer.cpp

```diff
@@ -48,31 +48,31 @@ Renderer::Renderer() :
     GL_CHECK_ERROR;
 }
 
-Renderer::~Renderer() {
-    ShaderProgramManager::destroyInstance();
-}
+Renderer::~Renderer() = default;
 
 void Renderer::initialize( uint width, uint height ) {
     /// For internal resources management in a filesystem
     auto resourcesRootDir {RadiumEngine::getInstance()->getResourcesDir()};
+    // Initialize managers
+    m_renderObjectManager  = RadiumEngine::getInstance()->getRenderObjectManager();
```

Comment:
+1 to not use raw pointers in new code.

Reply:
My understanding of isocpp documentation is that it is better to consider unique_ptr for such cases.

Reply:
No, unique_ptr is not adequate here. Nor shared ptr.
This pointer could be marked as `non-owning<Type>` as one may found in the [guidelines support library](https://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#S-gsl) but the GSL only define `owner<T>` as raw pointer are considered as non owning.

Here the pointer is own by the engine and (see the comment in the doc and i the code) :
from https://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#Ri-raw
> Every object passed as a raw pointer (or iterator) is assumed to be owned by the caller, so that its lifetime is handled by the caller. Viewed another way: ownership transferring APIs are relatively rare compared to pointer-passing APIs, so the default is “no ownership transfer.”
 
and from https://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#Rr-ptr
> R.3: A raw pointer (a T*) is non-owning

Here, the raw pointer is just an alias to a pointer owned by the Engine (a std::unique_ptr in the engine) that only the engine can delete and that is ensure to survive the renderer (lifetime of the engine is longer than lifetime of the Renderer)